code/game/objects/items/devices/scanners.dm

```diff
@@ -1006,21 +1006,24 @@ GENE SCANNER
 		symptomholder.Finalize()
 		symptomholder.Refresh()
 		to_chat(user, "<span class='warning'>you begin isolating [chosen].</span>")
-		if(do_mob(user, AM, (600 / scanner.rating)))
-			create_culture(symptomholder, user)
-	else if(do_mob(user, AM, (timer / scanner.rating)))
-		create_culture(A, user)
+		if(do_mob(user, AM, (600 / scanner.rating + 1)))
+			create_culture(symptomholder, user, AM)
+	else if(do_mob(user, AM, (timer / scanner.rating + 1)))
+		create_culture(A, user, AM)
 
 /obj/item/extrapolator/proc/create_culture(var/datum/disease/advance/A, mob/user)
-	if(cooldown > world.time - (1200 / scanner.rating))
+	if(cooldown > world.time - (12 / scanner.rating))
 		to_chat(user, "<span class='warning'>The extrapolator is still recharging!</span>")
 		return FALSE
+	if(!user.get_item_for_held_index(user.active_hand_index))
+		to_chat(user, "<span class='warning'>The extrapolator must be held in your active hand to work!</span>")
+		return FALSE
 	var/list/data = list("viruses" = list(A))
 	var/obj/item/reagent_containers/glass/bottle/B = new(user.loc)
 	cooldown = world.time
 	B.name = "[A.name] culture bottle"
 	B.desc = "A small bottle. Contains [A.agent] culture in synthblood medium."
 	B.reagents.add_reagent(/datum/reagent/blood, 20, data)
-	user.put_in_hands(B)
+	user.put_in_hand(B, user.get_inactive_hand_index(), TRUE)
```

Comment:
Why is this change being made? It seems weird and completely unnecessary to me, and might act weird with mobs with more than 2 hands, like grod.

Reply:
i suppose it need not be made- it was originally to jankily interrupt the do_mob before the current solution of checking hand